src/Passes/Source/Rules/Notation/Notation.hpp

```diff
@@ -0,0 +1,202 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/// @defgroup shorthandNotation Shorthand Notation
+
+#include "Rules/Notation/Call.ipp"
+#include "Rules/Notation/Phi.ipp"
```

Comment:
Would be nice to see a few words of comments in the code explaining what the `.ipp` file extension mean.

Reply:
`ipp` are implementational header files. They are like `.cpp` files but written in ipp-headers. They could as well have been `.hpp` files but normally you keep them in `.ipp` files to indicate that these contain the implementation details of a class. This allows you to keep `.hpp` headers pure definitions. I know that I do not keep a 100% to this in this PR, but the intention is that `.hpp` should always be pure definitions (with interface comments).

Reply:
Note `ipp` is usually used for template implementations.